File: 0007-reverse-integer/solution.py
class Solution:
    def reverse(self, x: int) -> int:
        n=x
        pos=0
        num=0
        
        if(n<0):
            pos=1
        n=abs(x)
        while n>0:
            num=num*10+n%10
            n=n//10
        if(pos):
            num=-abs(num)
        if(num< -2147483648 or num >2147483647):
            return 0
    
        
        return num

File: 0007-reverse-integer/solution.cpp
class Solution {
public:
    int reverse(int x) {
        
        cout<<INT_MIN<<" "<<INT_MAX;
        int flag=0;
        if(x<0)
        flag=1;
        int ans=abs(x);
        long res=0;
        while(ans>0){
            res=(res*10)+ans%10;
            ans/=10;
        }

        if(res<INT_MIN || res>INT_MAX)
        return 0;
        if(flag){
            return res-(2*res);
        }
        else{
            return res;
        }
        

        
    }
};
